File: facebook-common/src/main/java/com/facebook/LoginStatusCallback.java
/**
 * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
 *
 * You are hereby granted a non-exclusive, worldwide, royalty-free license to use,
 * copy, modify, and distribute this software in source code or binary form for use
 * in connection with the web services and APIs provided by Facebook.
 *
 * As with any software that integrates with the Facebook platform, your use of
 * this software is subject to the Facebook Developer Principles and Policies
 * [http://developers.facebook.com/policy/]. This copyright notice shall be
 * included in all copies or substantial portions of the software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
 * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
 * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
 * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
 * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */

package com.facebook;

import com.facebook.AccessToken;

/**
 * A callback class for getting the Login Status of a user.
 */
public interface LoginStatusCallback {

    /**
     * Called when an access token is able to be retrieved successfully.
     * @param accessToken The access token retrieved for the user
     */
    void onCompleted(final AccessToken accessToken);

    /**
     * Called when an access token could not be retrieved.
     */
    void onFailure();

    /**
     * Called when there was an error getting the login status of a user.
     * @param exception The error that occurred
     */
    void onError(final Exception exception);
}

File: facebook-common/src/main/java/com/facebook/LoginStatusCallback.kt
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 * All rights reserved.
 *
 * This source code is licensed under the license found in the
 * LICENSE file in the root directory of this source tree.
 */

package com.facebook

/** A callback class for getting the Login Status of a user. */
interface LoginStatusCallback {
  /**
   * Called when an access token is able to be retrieved successfully.
   *
   * @param accessToken The access token retrieved for the user
   */
  fun onCompleted(accessToken: AccessToken)

  /** Called when an access token could not be retrieved. */
  fun onFailure()

  /**
   * Called when there was an error getting the login status of a user.
   *
   * @param exception The error that occurred
   */
  fun onError(exception: Exception)
}
